Doc Laser Waist: GaussianBeam, PulseFrontTilt

ax3l opened the pull request:
This ~~changes~~ adds additional information to the the doc-string for the [beam waist](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71af8e8bf07d9fb36c33df561cc1f796133fc16f/src/picongpu/include/fields/laserProfiles/laserGaussianBeam.hpp#L138) W0 of the Gaussian Beam and the tilted version of it for laser implementations.
    
~~Before that, the in-focus intensity was still correct but the waist would have been to small if one was following the documentation. This is an old documentation bug that someone changed in pre-review (svn) times that slipped through.~~ The very short documentation before caused some confusion when trying to compare to experiments (for me :) ).

~~Users: if you are using the GaussianBeam profile in your simulation, please cross-check if the beam waist was set to what you intended it to be. It might be too small (resulting in a still correct in-focus intensity but smaller spot size and steeper divergence).~~

![laserwaist4](https://cloud.githubusercontent.com/assets/1353258/11837505/af2eecfa-a3e1-11e5-8b03-d81be6d91203.png)

(image: [v1](https://cloud.githubusercontent.com/assets/1353258/11030380/f1ad804a-86cd-11e5-94c2-334cf050fb66.png), ~~[v2](https://cloud.githubusercontent.com/assets/1353258/11211733/c1aca8f2-8d31-11e5-8453-eea6bcfa5df9.png)~~, [v3](https://cloud.githubusercontent.com/assets/1353258/11213215/0b601c64-8d3b-11e5-88b6-a1ba032568c5.png), [v4](https://cloud.githubusercontent.com/assets/1353258/11837505/af2eecfa-a3e1-11e5-8b03-d81be6d91203.png))


### Tasks

- [x] update examples (values kept, default LWFA slight increase for consistency)
- [x] check [wavepacket definition](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71af8e8bf07d9fb36c33df561cc1f796133fc16f/src/picongpu/include/fields/laserProfiles/laserWavepacket.hpp#L113-L116) -> wrong -> separate PR
- ~~[LaserPolynom](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71af8e8bf07d9fb36c33df561cc1f796133fc16f/src/picongpu/include/fields/laserProfiles/laserPolynom.hpp#L77)~~
- ~~write announcement on mailing list (after review)~~

PrometheusPi commented:
@ax3l great job. Please add your name to the license header wherever it is still missing.

PrometheusPi commented:
Equation (3) should be inverted:
$1/e^2 = e^-2:= E(a) / E(0)$

PrometheusPi commented:
For a correctly defined sigma you should use:
$ E(x) = E(0) * exp(- alpha (x/sigma_E)^2) $
with $ alpha = 1/2 $.
In equation (4) however, $ alpha = 1 $ is assumed.

PrometheusPi commented:
For you case of $ alpha = 1 $, the FWHM is:
$ sqrt(ln(2)) sigma_I = FWHM $

PrometheusPi commented:
There is something wrong with your definitions.
You use the standard deviation for normal distribution (alpha = 1/2) but use a non-normal distribution (alpha = 1) in the code. Please check this again. (I will add a WIP flag for now.)

ax3l commented:
there is a small missunderstanding what my *a* (not alpha) stands for, it's just an intermediate variable. will cross-check offline and post results.

ax3l commented:
@PrometheusPi Thanks. I flipped the typo, removed the confusing definition of "a" and updated the image to be more clear.

`E_x(...)` is a regular [Gaussian function](http://www.wolframalpha.com/input/?i=solve+1%2Fe+%3D+exp%28-0.5+*+%28x%2Fs%29^2%29%2Fexp%28-0.5+*+%280%2Fs%29^2%29+for+x), yes.

ax3l commented:
update: there is an other mess that I introduced, let me update one more

ax3l commented:
ok, clearified.

ax3l commented:
@PrometheusPi can be reviewed.

If I check the [definition of laser polynom](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71af8e8bf07d9fb36c33df561cc1f796133fc16f/src/picongpu/include/fields/laserProfiles/laserWavepacket.hpp#L113-L116) and compare it [with it's (copy-pasted) documentation](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71af8e8bf07d9fb36c33df561cc1f796133fc16f/src/picongpu/include/simulation_defines/param/laserConfig.param#L287-L290) it looks like it should be affected, too.

the [wavepacket documentation](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71af8e8bf07d9fb36c33df561cc1f796133fc16f/src/picongpu/include/simulation_defines/param/laserConfig.param#L227-L232) ([definition](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71af8e8bf07d9fb36c33df561cc1f796133fc16f/src/picongpu/include/fields/laserProfiles/laserWavepacket.hpp#L113-L116)) should probably say "where E goes to 1/e" and is therefore wrong in an other way.

PrometheusPi commented:
Your equations are correct. (I would use a different order in Eq. 2 for clarity in the following steps, but this is just my preference.)

PrometheusPi commented on examples/Bunch/include/simulation_defines/param/laserConfig.param:
```diff
@@ -56,11 +56,13 @@ BOOST_CONSTEXPR_OR_CONST float_64 AMPLITUDE_SI = _A0 * UNITCONV_A0_to_Amplitude_
 
-/** beam waist: distance from the axis where the pulse intensity (E^2)
+/** beam waist: distance from the axis where the electric field (E)
  *              decreases to its 1/e^2-th part,
```
according to your  first (not numbered) equation, at `x = w_0` the Electric field is at 1/e-th. The intensity is at 1/e^2-th.

ax3l replied:
oh it's actually documented correctly. the intensity is at 1/e^2 putting (4) into I and 1/e for E_x.

great! so I am just adding clearification here, no bug in the docs, yeah.

PrometheusPi commented on examples/Bunch/include/simulation_defines/param/laserConfig.param:
```diff
@@ -103,11 +105,13 @@ BOOST_CONSTEXPR_OR_CONST float_64 AMPLITUDE_SI = 1.738e13;
-    *  unit: meter */
-BOOST_CONSTEXPR_OR_CONST float_64 W0_SI = 4.246e-6;
+/** beam waist: distance from the axis where the electric field (E)
+ *              decreases to its 1/e^2-th part,
```
same here

PrometheusPi commented on examples/LaserWakefield/include/simulation_defines/param/laserConfig.param:
```diff
@@ -56,11 +56,13 @@ namespace picongpu
 
-            /** beam waist: distance from the axis where the pulse intensity (E^2)
+            /** beam waist: distance from the axis where the electric field (E)
              *              decreases to its 1/e^2-th part,
```
same here

PrometheusPi commented on examples/LaserWakefield/include/simulation_defines/param/laserConfig.param:
```diff
@@ -103,11 +105,13 @@ namespace picongpu
 
-            /** beam waist: distance from the axis where the pulse intensity (E^2)
+            /** beam waist: distance from the axis where the electric field (E)
              *              decreases to its 1/e^2-th part,
```
same here

PrometheusPi commented on examples/SingleParticleRadiationWithLaser/include/simulation_defines/param/laserConfig.param:
```diff
@@ -56,11 +56,13 @@ BOOST_CONSTEXPR_OR_CONST float_64 AMPLITUDE_SI = 1.738e13;
 
-/** beam waist: distance from the axis where the pulse intensity (E^2)
+/** beam waist: distance from the axis where the electric field (E)
  *              decreases to its 1/e^2-th part,
```
same here

PrometheusPi commented on examples/SingleParticleRadiationWithLaser/include/simulation_defines/param/laserConfig.param:
```diff
@@ -112,11 +114,13 @@ BOOST_CONSTEXPR_OR_CONST float_64 AMPLITUDE_SI = 1.738e13;
-    *  unit: meter */
-BOOST_CONSTEXPR_OR_CONST float_64 W0_SI = 4.246e-6;
+/** beam waist: distance from the axis where the electric field (E)
+ *              decreases to its 1/e^2-th part,
```
same here

PrometheusPi commented on src/picongpu/include/simulation_defines/param/laserConfig.param:
```diff
@@ -56,11 +56,13 @@ namespace picongpu
 
-            /** beam waist: distance from the axis where the pulse intensity (E^2)
+            /** beam waist: distance from the axis where the electric field (E)
              *              decreases to its 1/e^2-th part,
```
same here

PrometheusPi commented on src/picongpu/include/simulation_defines/param/laserConfig.param:
```diff
@@ -103,11 +105,13 @@ namespace picongpu
-                *  unit: meter */
-            BOOST_CONSTEXPR_OR_CONST float_64 W0_SI = 4.246e-6;
+            /** beam waist: distance from the axis where the electric field (E)
+             *              decreases to its 1/e^2-th part,
```
same here

PrometheusPi commented:
@ax3l Did you find time to check the equations and update the comments?

PrometheusPi commented:
@ax3l Did you find time to check the equations and update the comments?

ax3l commented:
but I think the definition of `wavepacket` is wrong: it should read `Intensity at 1/e` see [this definition](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71af8e8bf07d9fb36c33df561cc1f796133fc16f/src/picongpu/include/fields/laserProfiles/laserWavepacket.hpp#L113-L116): for W0 the wavepacket is for E at 1/sqrt(e)

PrometheusPi commented:
@ax3l Yes you are right. For the `wavepacket` the electric field is `e^(-0.5)=1/sqrt(e)` at `x=w_{0x}` thus at `x=w_{0x}` the Intensity `I=E^2=1/e`.

ax3l commented:
> wavepacket

ok, so I will update that in a separate PR to mark it as a bug and have
it separated.

PrometheusPi commented:
@ax3l we could either adjust the documentation **or** adjust the transversal envelop in the implementation

PrometheusPi commented:
@ax3l you are fast - yes make it a separate pull request please.

ax3l commented:
mail notification is strong with me.

PrometheusPi commented:
@ax3l For some reasons, you are not author of `examples/Bunch/include/simulation_defines/param/laserConfig.param`. Could you please add your name?

ax3l commented:
@PrometheusPi mergable :)
